File: svc-data/src/test/java/com/beancounter/marketdata/FigiAdapterTest.java
package com.beancounter.marketdata;

import static org.assertj.core.api.Assertions.assertThat;

import com.beancounter.common.model.Asset;
import com.beancounter.common.model.Currency;
import com.beancounter.common.model.Market;
import com.beancounter.marketdata.assets.figi.FigiAdapter;
import com.beancounter.marketdata.assets.figi.FigiAsset;
import org.junit.jupiter.api.Test;

public class FigiAdapterTest {
  @Test
  void is_CodePreserved() {
    FigiAdapter figiAdapter = new FigiAdapter();
    FigiAsset figiAsset = new FigiAsset("BRK", "BRK/B", "Mutual Fund");
    Asset asset = figiAdapter.transform(
        new Market("TEST", new Currency("USD")),
        "BRK.B",
        figiAsset
    );

    assertThat(asset)
        .hasFieldOrPropertyWithValue("name", "BRK")
        .hasFieldOrPropertyWithValue("code", "BRK.B")
        .hasFieldOrPropertyWithValue("category", "Mutual Fund");
  }
}

File: svc-data/src/test/kotlin/com/beancounter/marketdata/FigiAdapterTest.kt
package com.beancounter.marketdata

import com.beancounter.common.model.Market
import com.beancounter.marketdata.Constants.Companion.USD
import com.beancounter.marketdata.assets.figi.FigiAdapter
import com.beancounter.marketdata.assets.figi.FigiAsset
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

/**
 * Test Bloomberg Figi transformations.
 */
class FigiAdapterTest {
    @Test
    fun is_CodePreserved() {
        val figiAdapter = FigiAdapter()
        val figiAsset = FigiAsset("BRK", "BRK/B", "Mutual Fund")
        val asset = figiAdapter.transform(
            Market("TEST", USD),
            "BRK.B",
            figiAsset
        )
        Assertions.assertThat(asset)
            .hasFieldOrPropertyWithValue("name", "BRK")
            .hasFieldOrPropertyWithValue("code", "BRK.B")
            .hasFieldOrPropertyWithValue("category", "Mutual Fund")
    }
}
